sql/src/main/java/org/apache/druid/sql/calcite/planner/QueryUtils.java

```diff
@@ -55,4 +57,13 @@ public static List<ColumnMapping> buildColumnMappings(
 
     return columnMappings;
   }
+
+  public static JoinAlgorithm getJoinAlgorithm(Join join, PlannerContext plannerContext)
+  {
+    if (join.getHints().isEmpty()) {
+      return plannerContext.getJoinAlgorithm();
+    }
+
+    return JoinHint.fromString(join.getHints().get(0).hintName).getJoinAlgorithm();
```

Comment:
this may work right now but it assumes that there is only one hint and that may only be a `JoinHint`